### PR TITLE
api/server: add brick creation checks in nic_add

### LINKED ISSUE #18: Graph::nic_add don't check brick creation fail
Graph::nic_add create a lot of new brick, using shared_ptr, but there is no check to know if
the creation has been successful.
we should use that: http://www.cplusplus.com/reference/memory/shared_ptr/operator%20bool/

### PR DESCRIPTION
**Pull request by outscale-jju**
closes #18

Signed-off-by: Jerome Jutteau <jerome.jutteau@outscale.com>

### CODE REVIEW

File: api/server/graph.cc [hunk 1]
```diff
@@ -356,6 +360,10 @@ std::string Graph::nic_add(const app::Nic &nic) {
     name = "vhost-" + gn.id;
     gn.vhost = Brick(Pg::vhost_new(name.c_str(), 1, 1, EAST_SIDE),
                      Pg::destroy);
+    if (!gn.vhost) {
+        LOG_ERROR_("Vhost-user creation failed");
```
Comment (outscale-mgo) [inline api/server/graph.cc]: same here I would use `LOG_ERROR_("%s creation failed", name.c_str());`

File: api/server/graph.cc [hunk 2]
```diff
@@ -344,6 +344,10 @@ std::string Graph::nic_add(const app::Nic &nic) {
     nic.mac.bytes(mac.ether_addr_octet);
     gn.antispoof = Brick(Pg::antispoof_new(name.c_str(), WEST_SIDE, mac),
                          Pg::destroy);
+    if (!gn.antispoof) {
+        LOG_ERROR_("Antispoof creation failed");
```
Comment (outscale-mgo) [inline api/server/graph.cc]: I would use the brick name

File: api/server/graph.cc [hunk 3]
```diff
@@ -364,6 +372,11 @@ std::string Graph::nic_add(const app::Nic &nic) {
                                      PG_PRINT_FLAG_CLOSE_FILE,
                                      NULL),
                        Pg::destroy);
+    if (!gn.sniffer) {
+        LOG_ERROR_("Sniffer creation failed");
```
Comment (outscale-mgo) [inline api/server/graph.cc]: same

File: api/server/graph.cc [hunk 4]
```diff
@@ -383,6 +396,10 @@ std::string Graph::nic_add(const app::Nic &nic) {
         name = "switch-" + std::to_string(nic.vni);
         vni.sw = Brick(Pg::switch_new(name.c_str(),
                                       1, 30, EAST_SIDE), Pg::destroy);
+        if (!vni.sw) {
+            LOG_ERROR_("Switch creation failed");
```
Comment (outscale-mgo) [inline api/server/graph.cc]: same

File: api/server/graph.cc (hunk 2 above)
Comment (outscale-jju) [inline api/server/graph.cc]: fixed

File: api/server/graph.cc (hunk 1 above)
Comment (outscale-jju) [inline api/server/graph.cc]: fixed

File: api/server/graph.cc (hunk 3 above)
Comment (outscale-jju) [inline api/server/graph.cc]: fixed

File: api/server/graph.cc (hunk 4 above)
Comment (outscale-jju) [inline api/server/graph.cc]: fixed

**Pull request closed without merging**